File: src/main/java/victor/training/cleancode/SplitLoop.java
package victor.training.cleancode;

import lombok.Builder;

import java.util.List;

public class SplitLoop {
  // TODO Split loops and refactor to .stream. Run Tests✅
  public String computeStats(List<Employee> employees) {
    // daca faci .forEach(e-> {
    // mult cod
    // };
    // performanta nu va durea 1 for -> 2 for -> 2 .stream()
    double totalConsultantSalary = employees.stream()
        .filter(Employee::consultant)
        .mapToDouble(Employee::salary)
        .sum();
    List<Integer> employeeIds = employees.stream().map(Employee::id).toList();

    System.out.println("Employee IDs: " + employeeIds);
    return "Total consultant salary: " + totalConsultantSalary + "; ids: " + employeeIds;
  }

  @Builder(toBuilder = true)
  public record Employee(
      Integer id,
      int age,
      Integer salary,
      boolean consultant) {
  }
}

File: src/main/java/victor/training/cleancode/SplitLoop.py
from dataclasses import dataclass
from typing import List

# TODO Split loops and refactor to .stream. Run Tests✅
def compute_stats(employees: List["Employee"]) -> str:
    employee_ids = []
    total_consultant_salary = 0
    for employee in employees:
        if employee.consultant:
            total_consultant_salary += employee.salary
        employee_ids.append(employee.id)
    print("Employee IDs: " + str(employee_ids))
    return f"Total consultant salary: {total_consultant_salary}; ids: {employee_ids}"

@dataclass
class Employee:
    id: int
    age: int
    salary: int
    consultant: bool